rclpy/rclpy/impl/object_proxy.py

```diff
@@ -0,0 +1,165 @@
+        'iter', 'oct', 'hex'
+    ]:
+        code_object = compile(
+            "def __{0}__(self): return {0}(self.__actual__)".format(name),
+            __file__,
+            'exec',
+            ast.PyCF_ONLY_AST,
+        )
+        ast.increment_lineno(code_object, inspect.getframeinfo(inspect.currentframe()).lineno - 6)
+        exec(compile(code_object, __file__, 'exec'))
+
+    for name in ['cmp', 'coerce', 'divmod']:
+        code_object = compile(
+            "def __{0}__(self, ob): return {0}(self.__actual__, ob)".format(name),
+            __file__,
+            'exec',
+            ast.PyCF_ONLY_AST,
+        )
+        ast.increment_lineno(code_object, inspect.getframeinfo(inspect.currentframe()).lineno - 6)
+        exec(compile(code_object, __file__, 'exec'))
+
+    for name, op in [
+        ('lt', '<'), ('gt', '>'), ('le', '<='), ('ge', '>='), ('eq', '=='), ('ne', '!=')
+    ]:
+        code_object = compile(
+            "def __{0}__(self, ob): return self.__actual__ {1} ob".format(name, op),
+            __file__,
+            'exec',
+            ast.PyCF_ONLY_AST,
+        )
+        ast.increment_lineno(code_object, inspect.getframeinfo(inspect.currentframe()).lineno - 6)
+        exec(compile(code_object, __file__, 'exec'))
+
+    for name, op in [('neg', '-'), ('pos', '+'), ('invert', '~')]:
+        code_object = compile(
+            "def __{0}__(self): return {1} self.__actual__".format(name, op),
+            __file__,
+            'exec',
+            ast.PyCF_ONLY_AST,
+        )
+        ast.increment_lineno(code_object, inspect.getframeinfo(inspect.currentframe()).lineno - 6)
+        exec(compile(code_object, __file__, 'exec'))
+
+    for name, op in [
+        ('or', '|'), ('and', '&'), ('xor', '^'), ('lshift', '<<'), ('rshift', '>>'),
+        ('add', '+'), ('sub', '-'), ('mul', '*'), ('div', '/'), ('mod', '%'),
+        ('truediv', '/'), ('floordiv', '//')
+    ]:
+        code_object = compile(
+            """
+def __{name}__(self,ob):
+    return self.__actual__ {op} ob
+
+def __r{name}__(self,ob):
+    return ob {op} self.__actual__
+
+def __i{name}__(self,ob):
+    self.__actual__ {op}=ob
+    return self
+""".format(name=name, op=op),
+            __file__,
+            'exec',
+            ast.PyCF_ONLY_AST,
+        )
+        ast.increment_lineno(code_object, inspect.getframeinfo(inspect.currentframe()).lineno - 6)
+        exec(compile(code_object, __file__, 'exec'))
+
+    del name, op
+
+    def __rdivmod__(self, ob):
+        return divmod(ob, self.__actual__)
+
+    def __pow__(self, *args):
+        return pow(self.__actual__, *args)
+
+    def __ipow__(self, ob):
+        self.__actual__ **= ob
+        return self
+
+    def __rpow__(self, ob):
+        return pow(ob, self.__actual__)
+
+
+class ObjectProxy(AbstractProxy):
+    """Proxy for an internally stored object."""
+
+    __slots__ = "__actual__"
+
+    def __init__(self, subject):
+        self.__actual__ = subject
```

Comment:
The class is based on an existing proxy object solution for Python that is generic. Since it will be used throughout the codebase I don't know what someone might want to do to the implementation module other than access members, but I figured I'd give my best shot at covering them all. Also by covering more surface area with the `ObjectProxy` class, it makes it something that can be reused by others in `rclpy` if they need to. I don't see any reason to remove part of the functionality here.